tests/keras_contrib/callbacks/dead_relu_detector_test.py

```diff
@@ -26,16 +37,112 @@ def do_test(weights, expected_warnings, verbose):
                 assert "dead neurons" in str(warn_item.message)
 
-    weights_1_dead = np.ones((1, 10))      # weights that correspond to NN with 1/10 neurons dead
+    weights_1_dead = np.ones(shape_weights)  # weights that correspond to NN with 1/10 neurons dead
+    weights_2_dead = np.ones(shape_weights)  # weights that correspond to NN with 2/10 neurons dead
+
     weights_1_dead[:, 0] = 0
-    weights_2_dead = np.ones((1, 10))      # weights that correspond to NN with 2/10 neurons dead
-    weights_2_dead[:, 0] = 0
-    weights_2_dead[:, 1] = 0
+    weights_2_dead[:, 0:2] = 0
 
     do_test(weights_1_dead, verbose=True, expected_warnings=1)
     do_test(weights_1_dead, verbose=False, expected_warnings=0)
     do_test(weights_2_dead, verbose=True, expected_warnings=1)
 
 
+def test_DeadDeadReluDetector_bias():
+    n_samples = 9
+
+    input_shape = (n_samples, 4)  # 4 input features
+    shape_weights = (4, 10)
+    shape_out = (n_samples, 10)  # 10 output features
+    shape_bias = (10, )
+
+    # ignore batch size
+    input_shape_dense = tuple(input_shape[1:])
+
+    def do_test(weights, bias, expected_warnings, verbose):
+        with warnings.catch_warnings(record=True) as w:
+            dataset = np.ones(input_shape)  # data to be fed as training
+            model = Sequential()
+            model.add(Dense(10, activation='relu', input_shape=input_shape_dense,
+                            use_bias=True, weights=[weights, bias], name='dense'))
+            model.compile(optimizer='sgd', loss='categorical_crossentropy')
+            # model.compile(optimizer=None, loss='categorical_crossentropy')
+            model.fit(
+                dataset,
+                np.ones(shape_out),
+                epochs=1,
+                callbacks=[callbacks.DeadReluDetector(dataset, verbose=verbose)],
+                verbose=False
+            )
+            assert len(w) == expected_warnings
+            for warn_item in w:
+                assert issubclass(warn_item.category, RuntimeWarning)
+                assert "dead neurons" in str(warn_item.message)
+
+    weights_1_dead = np.ones(shape_weights)  # weights that correspond to NN with 1/10 neurons dead
+    weights_2_dead = np.ones(shape_weights)  # weights that correspond to NN with 2/10 neurons dead
+
+    weights_1_dead[:, 0] = 0
+    weights_2_dead[:, 0:2] = 0
+
+    bias = np.zeros(shape_bias)
+
+    do_test(weights_1_dead, bias, verbose=True, expected_warnings=1)
+    do_test(weights_1_dead, bias, verbose=False, expected_warnings=0)
+    do_test(weights_2_dead, bias, verbose=True, expected_warnings=1)
+
+
+def test_DeadDeadReluDetector_conv():
+    n_samples = 9
+
+    # (5, 5) kernel, 4 input featuremaps and 10 output featuremaps
+    if K.backend() == 'tensorflow':
+        input_shape = (n_samples, 5, 5, 4)
+    elif K.backend() == 'theano':
+        input_shape = (n_samples, 4, 5, 5)
+    else:
+        raise ValueError('Unknown backend: {}'.format(K.backend()))
```

Comment:
@infelane 
not good to rely on exact backend implementation

Reply:
K.image_data_format() should do the trick
Done